docs/architecture/adr-054-crypto-encoding-2.md

```diff
@@ -0,0 +1,67 @@
+# ADR 054: Crypto encoding (part 2)
+
+## Changelog
+
+\*2020-2-27: Created
+
+## Context
+
+Amino has been a pain point of many users in the ecosystem. While Tendermint does not suffer greatly from the performance degradation introduced by amino, we are making an effort in moving the encoding format to a widely adopt format, [Protocol Buffers](https://developers.google.com/protocol-buffers). With this migration a new standard is needed for the encoding of keys. This will cause ecosystem wide breaking changes.
+
+Currently amino encodes keys as `<PrefixBytes> <Length> <ByteArray>`.
+
+> With keys becoming a [oneof type](https://developers.google.com/protocol-buffers/docs/proto3#oneof) we have the option of removing the prefix bytes.
+
+## Options
+
+If there are more options please leave a comment and we can discuss it.
+
+1. Remove backwards compatibility and go with `oneof` proto encoding of keys.
+
+- This will be a major breaking change in the ecosystem.
+- Will need a migration script to be released in conjunction with the release which will have this change.
```

Comment:
IMHO this is the cleanest option. To go this way we'd need to make sure that we do not use the amino encoded bytes anywhere user facing (e.g. when bechifying: are we using the raw bytes or the prefixed bytes?).  Even if there are cases where this is user facing, it might be worth exploring.